File: Minimize/Python/test01.py
#!/usr/bin/env python3

import sys,math
import numpy as np
import rupert

def main(argc,argv):
  mat1 = np.zeros(9).reshape(3,3)
  mat2 = np.zeros(9).reshape(3,3)
  mat3 = np.zeros(9).reshape(3,3)
  mat4 = np.zeros(9).reshape(3,3)
  mat5 = np.zeros(9).reshape(3,3)

  print("Starting to write Python code to work on the Generalized Rupert Problem.")
  print()
  
  a01 = math.pi/4
  a02 = 0.0
  a12 = 4 * math.atan(math.sqrt(5.0 - 2.0*math.sqrt(6.0)))

  print("a01: %9.6f (%9.4f)" % (a01,math.degrees(a01)))
  print("a02: %9.6f (%9.4f)" % (a02,math.degrees(a02)))
  print("a12: %9.6f (%9.4f)" % (a12,math.degrees(a12)))
  print()

  rot = [a01,a02,a12]

  rup = rupert.rupert(2,3,rot)

  print(rup.mat)
  print()

  print(np.linalg.det(rup.mat))
  print()

  print(rup.value())
  print()


if __name__ == "__main__":
  main(len(sys.argv),sys.argv)

File: Minimize/C++/src/test01.cpp
#include <cstdio>
#include <cmath>

#include "angle.h"
#include "matrix.h"

int main(int argc, char **argv)
{
  MATRIX mat1(3,3);
  MATRIX mat2(3,3);
  MATRIX mat3(3,3);
  MATRIX mat4(3,3);
  MATRIX mat5(3,3);

  double a01,a02,a12;

  double det;

  printf("Starting to write C code to work on the Generalized Rupert Problem.\n");
  printf("\n");

  a01 = M_PI/4;
  a02 = 0.0;
  a12 = 4 * atan(sqrt(5-2*sqrt(6)));

  printf("a01: %9.6f (%9.4f)\n",a01,rad2deg(a01));
  printf("a02: %9.6f (%9.4f)\n",a02,rad2deg(a02));
  printf("a12: %9.6f (%9.4f)\n",a12,rad2deg(a12));
  printf("\n");

  mat4.identity();
  mat4.print();
  printf("\n");

  mat4.rotate(a12,1,2);
  mat4.print();
  printf("\n");

  mat4.rotate(a02,0,2);
  mat4.print();
  printf("\n");

  mat4.rotate(a01,0,1);
  mat4.print();
  printf("\n");

  return 0;
}
